string FileName
float64 radius
float64 dx
float64 dy
float64 lower_level
float64 upper_level
---